src/zeroconf/_services/info.py

```diff
@@ -431,46 +431,49 @@ def _process_record_threadsafe(self, zc: 'Zeroconf', record: DNSRecord, now: flo
         if record.is_expired(now):
             return False
 
-        if record.key == self.server_key and isinstance(record, DNSAddress):
+        record_key = record.key
+        if record_key == self.server_key and type(record) is DNSAddress:
```

Comment:
These are all internal objects and should never be subclasseses so `type` is faster here